File: app/src/main/java/com/imangazalievm/bubbble/domain/userprofile/UserShotsInteractor.java
package com.imangazalievm.bubbble.domain.userprofile;


import com.imangazalievm.bubbble.domain.global.models.Shot;
import com.imangazalievm.bubbble.domain.global.models.User;
import com.imangazalievm.bubbble.domain.global.models.UserShotsRequestParams;
import com.imangazalievm.bubbble.domain.global.repositories.ShotsRepository;
import com.imangazalievm.bubbble.domain.global.repositories.UsersRepository;
import com.imangazalievm.bubbble.presentation.mvp.global.SchedulersProvider;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserShotsInteractor {

    private UsersRepository usersRepository;
    private ShotsRepository shotsRepository;
    private SchedulersProvider schedulersProvider;

    @Inject
    public UserShotsInteractor(
            UsersRepository usersRepository,
            ShotsRepository shotsRepository,
            SchedulersProvider schedulersProvider
    ) {
        this.usersRepository = usersRepository;
        this.shotsRepository = shotsRepository;
        this.schedulersProvider = schedulersProvider;
    }

    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId);
    }

    public Single<List<Shot>> getUserShots(UserShotsRequestParams requestParams) {
        return shotsRepository.getUserShots(requestParams)
                .subscribeOn(schedulersProvider.io());
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/domain/userprofile/UserShotsInteractor.kt
package com.imangazalievm.bubbble.domain.userprofile

import com.imangazalievm.bubbble.data.users.UsersRepository
import com.imangazalievm.bubbble.data.shots.ShotsRepository
import com.imangazalievm.bubbble.presentation.global.SchedulersProvider
import com.imangazalievm.bubbble.domain.global.models.UserShotsRequestParams
import com.imangazalievm.bubbble.domain.global.models.Shot
import com.imangazalievm.bubbble.domain.global.models.User
import io.reactivex.Single
import javax.inject.Inject

class UserShotsInteractor @Inject constructor(
    private val usersRepository: UsersRepository,
    private val shotsRepository: ShotsRepository
) {

    suspend fun getUser(userId: Long): User {
        return usersRepository.getUser(userId)
    }

    suspend fun getUserShots(requestParams: UserShotsRequestParams): List<Shot> {
        return shotsRepository.getUserShots(requestParams)
    }

}